alodataset/kitti_odometry.py

```diff
@@ -0,0 +1,273 @@
+import os
+import torch
+import numpy as np
+from typing import List, Dict, Any
+
+from alodataset import BaseDataset, SplitMixin, Split
+from aloscene import Frame, Pose, CameraIntrinsic
+
+
+def sequence_index(start, seq_size, skip):
+    end = start + (seq_size - 1) * (skip + 1)
+    return np.linspace(start, end, seq_size).astype(int).tolist()
+
+
+def sequence_indices(n_samples, seq_size, skip, seq_skip):
+    for start in range(0, n_samples - (seq_size - 1) * (skip + 1), seq_skip + 1):
+        yield sequence_index(start, seq_size, skip)
+
+
+class KittiOdometryDataset(BaseDataset, SplitMixin):
+    def __init__(
+        self,
+        name="kitti_odometry",
+        sequences=None,
+        grayscale=True,
+        right_frame=True,
+        sequence_size=3,
+        skip=1,
+        sequence_skip=1,
+        **kwargs,
+    ):
+        super().__init__(name=name, **kwargs)
+
+        self.grayscale = grayscale
+        self.right_frame = right_frame
+        self.sequence_size = sequence_size
+        self.skip = skip
+        self.sequence_skip = sequence_skip
+
+        if self.sample:
+            return
+
+        self.dataset_dir = os.path.join(self.dataset_dir, "dataset")
+
+        self.sequences = self._load_sequences(sequences)
+
```

Comment:
you should check that the selected sequences are all compatible with `self.split`